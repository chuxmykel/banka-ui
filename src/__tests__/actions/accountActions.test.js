import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import nock from 'nock';
import {
  getAccounts,
  getAccountTransactions,
  createAccount,
  getAllUsersAccounts,
  changeAccountStatus,
  deleteAccount,
  transaction,
} from '@Actions/accountActions';

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);
const url = 'https://a-bank.herokuapp.com/api/v1';

describe('Account action tests', () => {
  describe('Account Action tests return values', () => {
    const payload = {
      accountType: 'savings',
      initialDeposit: 50000,
    };

    const accountNumber = '1234567890';

    it('getAccounts should return expected values', () => {
      expect(getAccounts()).toMatchSnapshot();
    });

    it('getAccountTransactions should return expected values', () => {
      expect(getAccountTransactions(accountNumber)).toMatchSnapshot();
    });

    it('createAccount should return expected values', () => {
      expect(createAccount(payload)).toMatchSnapshot();
    });

    it('getAllUsersAccounts should return expected values', () => {
      expect(getAllUsersAccounts()).toMatchSnapshot();
    });

    it('changeAccountStatus should return expected values', () => {
      expect(changeAccountStatus(accountNumber)).toMatchSnapshot();
    });

    it('deleteAccount should return expected values', () => {
      expect(deleteAccount(accountNumber)).toMatchSnapshot();
    });

    it('transaction should return expected values', () => {
      expect(transaction(accountNumber)).toMatchSnapshot();
    });
  });

  describe('Async action creators test', () => {
    let store;
    const response = {
      data: {
        accounts: [],
        data: [],
      },
    };

    beforeEach(() => {
      store = mockStore({});
    });

    afterEach(() => {
      nock.cleanAll();
    });

    it('fetches accounts successfully', () => {
      nock(url)
        .persist()
        .get('/user/accounts')
        .reply(200, response);

      return store.dispatch(getAccounts())
        .then(() => {
          expect(store.getActions()).toMatchSnapshot();
        });
    });

    it('fetches transactions successfully', () => {
      const accountNumber = '1234567890';
      nock(url)
        .get(`/accounts/${accountNumber}/transactions`)
        .reply(200, response);

      return store.dispatch(getAccountTransactions(accountNumber))
        .then(() => {
          expect(store.getActions()).toMatchSnapshot();
        });
    });

    it('creates accounts successfully', () => {
      const accountDetails = {
        accountType: 'savings',
        initialDeposit: 50000,
      };

      nock(url)
        .post('/accounts')
        .reply(201);

      return store.dispatch(createAccount(accountDetails))
        .then(() => {
          expect(store.getActions()).toMatchSnapshot();
        });
    });
  });
});
