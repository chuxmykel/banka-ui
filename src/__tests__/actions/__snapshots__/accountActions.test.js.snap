// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Account action tests Account Action tests return values changeAccountStatus should return expected values 1`] = `[Function]`;

exports[`Account action tests Account Action tests return values createAccount should return expected values 1`] = `[Function]`;

exports[`Account action tests Account Action tests return values deleteAccount should return expected values 1`] = `[Function]`;

exports[`Account action tests Account Action tests return values getAccountTransactions should return expected values 1`] = `[Function]`;

exports[`Account action tests Account Action tests return values getAccounts should return expected values 1`] = `[Function]`;

exports[`Account action tests Account Action tests return values getAllUsersAccounts should return expected values 1`] = `[Function]`;

exports[`Account action tests Account Action tests return values transaction should return expected values 1`] = `[Function]`;

exports[`Account action tests Async action creators test creates accounts successfully 1`] = `
Array [
  Object {
    "type": "LOADING",
  },
  Object {
    "type": "LOADING",
  },
  Object {
    "payload": "createAccount",
    "type": "MODAL_CLOSE",
  },
  Object {
    "type": "NOT_ LOADING",
  },
]
`;

exports[`Account action tests Async action creators test fetches accounts successfully 1`] = `
Array [
  Object {
    "type": "LOADING",
  },
  Object {
    "payload": undefined,
    "type": "GET_USER_ACCOUNTS",
  },
  Object {
    "type": "NOT_ LOADING",
  },
]
`;

exports[`Account action tests Async action creators test fetches transactions successfully 1`] = `
Array [
  Object {
    "type": "LOADING",
  },
  Object {
    "payload": Object {
      "accounts": Array [],
      "data": Array [],
    },
    "type": "GET_ACCOUNT_TRANSACTIONS",
  },
  Object {
    "type": "NOT_ LOADING",
  },
]
`;
